### ISSUE TITLE
Segfault in HSFrameLeaf::split()

### ISSUE THREAD

**Issue by ypnos** [herbstluftwm/herbstluftwm#167]
Doing a exploding split with two windows. It turns out that the argument `int childrenLeaving` is -1. This leads to a segfault in line 938:

    vector<HSClient*> leaves(clients.begin() + childrenStaying, clients.end());

because this goes wrong first:

    int childrenStaying = std::max((size_t)0, clients.size() - childrenLeaving);

childrenStaying is maxed to clients.size() + 1.
But why do we get -1 as parameter in the first place?
This problem might have been introduced in 056f522cb6d39eaf1d15455454a3e628f62135f8.

```
    size_t count1 = 0;
    if (exploding) {
        count1 -= (frame->clientCount() + 1) / 2;      // new count for the first frame
    }
```

In the exploding case, count1 appears to always be negative (overflown, that is).

**Comment by t-wissmann** [comment]
There are some things wrong indeed. This variable should be called `childrenLeaving` or `count2` and not `count1`, because it tells the number of clients that go into the second frame. This is why it is `0` for all the cases except `explode`. For `explode`, at most half the clients should go to the second frame, so your last snippet should be as follows:
```
    size_t childrenLeaving = 0;
    if (exploding) {
         childrenLeaving = frame->clientCount() / 2;
    }
```

**Comment by ypnos** [comment]
Thank you for the clarification as I didn't really get the point of that code piece at all. Would you like me to write a fix or do you prefer to fix it yourself?

**Comment by t-wissmann** [comment]
I'll take care of it.

**Comment by t-wissmann** [comment]
I've fixed this and other minor things in the PR #174

**Comment by ypnos** [comment]
Fixed in 4b64956fa4dc0f7c817973cbe491571626f2d6f4
